Model/Sync/Catalog/MagentoProductToYotpoProductAdapter.php

```diff
@@ -0,0 +1,329 @@
+<?php
+
+namespace Yotpo\Core\Model\Sync\Catalog;
+
+use Magento\Framework\UrlInterface;
+use Magento\Catalog\Model\Product;
+use Magento\CatalogInventory\Model\StockRegistry;
+use Yotpo\Core\Model\Config as YotpoCoreConfig;
+
+/**
+ * Class Logger - For customized logging
+ */
+class MagentoProductToYotpoProductAdapter
+{
+
+    const GTIN_TYPE_TO_GETTER_METHOD_NAME_MAP = [
+        'EAN' => 'getEan',
+        'UPC' => 'getUpc',
+        'ISBN' => 'getIsbn'
+    ];
+
+    const CUSTOM_ATTRIBUTE_TO_GETTER_METHOD_NAME_MAP = [
+        'is_blocklisted' => 'getIsBlocklisted',
+        'review_form_tag' => 'getReviewFormTag'
+    ];
+
+    /**
+     * @var StockRegistry
+     */
+    protected $stockRegistry;
+
+    /**
+     * @var YotpoCoreConfig
+     */
+    protected $yotpoCoreConfig;
+
+    /**
+     * @param StockRegistry $stockRegistry
+     * @param YotpoCoreConfig $yotpoCoreConfig
+     */
+    public function __construct(
+        StockRegistry $stockRegistry,
+        YotpoCoreConfig $yotpoCoreConfig
+    ) {
+        $this->stockRegistry = $stockRegistry;
+        $this->yotpoCoreConfig = $yotpoCoreConfig;
+    }
+
+    /**
+     * @param Product $item
+     * @return array
+     */
+    public function adapt(Product $item) {
+        $yotpoProduct = [];
+
+        $yotpoProduct['row_id'] = $this->getRowId($item);
+        $yotpoProduct['external_id'] = $this->getExternalId($item);
+        $yotpoProduct['name'] = $this->getName($item);
```

Comment:
I don't see that the `name` was set at the old method usage. Is it new? was it already set before calling the attribute mapping? If I'm missing it can you point me to where it was set before?

Reply:
It's here - https://github.com/YotpoLtd/magento2-module-core/pull/216/files#diff-34b81996b689eea325a28f6d3296417b0c6c518b498207c3d171a5970454e6dcL65 :)

Reply:
I meant in the code, but I know see the 'default' if clause which uses it. Thanks.